ledger/src/blockstore.rs

```diff
@@ -8776,26 +8773,6 @@ pub mod tests {
         }
     }
 
-    #[test]
-    fn test_remove_shred_data_complete_flag() {
-        let ledger_path = get_tmp_ledger_path_auto_delete!();
-        let blockstore = Blockstore::open(ledger_path.path()).unwrap();
-
-        let (mut shreds, entries) = make_slot_entries(0, 0, 1);
-
-        // Remove the data complete flag from the last shred
-        shreds[0].unset_data_complete();
-
-        blockstore.insert_shreds(shreds, None, false).unwrap();
-
-        // Check that the `data_complete` flag was unset in the stored shred, but the
-        // `last_in_slot` flag is set.
-        let stored_shred = &blockstore.get_data_shreds_for_slot(0, 0).unwrap()[0];
-        assert!(!stored_shred.data_complete());
-        assert!(stored_shred.last_in_slot());
-        assert_eq!(entries, blockstore.get_any_valid_slot_entries(0, 0));
-    }
```

Comment:
@carllin https://github.com/solana-labs/solana/pull/13626 added this test to address https://github.com/solana-labs/solana-security/issues/14 
It seems like that we no longer need this test because the shred cannot be `LAST_SHRED_IN_SLOT` without being `DATA_COMPLETE_SHRED` as well.
The test currently fails because when you call `Shred::unset_data_complete` the flags become invalid and so `Shred::sanitize` fails which will prevent `Blockstore::insert_shreds` from inserting the shred.

Please double check and confirm that we no longer need test.

Reply:
This makes sense, we can remove the test. Thanks for making this logic much tighter!